core/new-gui/src/app/workspace/component/navigation/navigation.component.ts

```diff
@@ -434,6 +437,44 @@ export class NavigationComponent {
   onClickGetAllVersions() {
     this.workflowVersionService.clickDisplayWorkflowVersions();
   }
+
+  private handleVersionDisplay(): void {
+    this.workflowVersionService
+      .getDisplayParticularVersionStream()
+      .pipe(untilDestroyed(this))
+      .subscribe(displayVersionFlag => {
+        this.particularVersionDate =
+          this.workflowActionService.getWorkflowMetadata().creationTime === undefined
+            ? ""
+            : "" +
+              this.datePipe.transform(
+                this.workflowActionService.getWorkflowMetadata().creationTime,
+                "MM/dd/yyyy HH:mm:ss zzz",
+                Intl.DateTimeFormat().resolvedOptions().timeZone,
+                "en"
+              );
+        this.displayParticularWorkflowVersion = displayVersionFlag;
+      });
+  }
+
+  closeParticularVersionDisplay() {
+    this.displayParticularWorkflowVersion = false;
+    this.workflowActionService.enableWorkflowModification();
+    // reload the old workflow don't persist anything
+    this.workflowActionService.reloadWorkflow(this.workflowActionService.getTempWorkflow());
+    this.workflowActionService.resetTempWorkflow();
+    this.workflowActionService.setWorkflowPersistFlag(true);
+  }
+
+  revertToVersion() {
```

Comment:
I find `closeParticularVersionDisplay` and `revertToVersion` two functions, which are two different ways to exit out from "particularVersionDisplay" mode. However, I do not find (at least explicitly) where you enter the "particularVersionDisplay" mode? Especially, I want to confirm the `persistWorkflowFlag` is turned off in such enter function.

Reply:
moved `close` `revert` and `display` to `WorkflowVersionService` so the logic is in one place